spokestack/tts/clients/spokestack.py

```diff
@@ -0,0 +1,113 @@
+"""
+This module contains the Spokestack client for text to speech
+"""
+import base64
+import hashlib
+import hmac
+import json
+
+import requests
+
+
+MODES = {
+    "ssml": "synthesizeSSML",
+    "markdown": "synthesizeMarkdown",
+    "text": "synthesizeText",
+}
+
+
+class TextToSpeechClient:
+    """ Spokestack Text to Speech Client
+
+    Args:
+        key_id (str):
+        key_secret (str):
+        url (str):
+    """
+
+    def __init__(self, key_id: str, key_secret: str, url: str) -> None:
+
+        self._key_id = key_id
+        self._key = key_secret.encode("utf-8")
+        self._url = url
+
+    def synthesize_speech(
+        self, utterance: str, mode: str = "text", voice: str = "demo-male"
+    ) -> bytes:
+        """ Converts the given utterance to speech
+
+        Args:
+            utterance (str):
+            mode (str):
+            voice (str):
+
+        Returns: Encoded Audio
+
+        """
+        body = self._build_body(utterance, mode, voice)
+        signature = base64.b64encode(
+            hmac.new(self._key, body.encode("utf-8"), hashlib.sha256).digest()
+        ).decode("utf-8")
+        headers = {
+            "Authorization": f"Spokestack {self._key_id}:{signature}",
+            "Content-Type": "application/json",
+        }
+
+        response = requests.post(self._url, headers=headers, data=body).json()
```

Comment:
you want to use `stream=True` here and return `response.iter_content`, so that you don't wait until all of the audio has been synthesized and downloaded before you return to the client
this also means that the typespec would change

Reply:
this code is correctly checking for graphql errors, but you'll also want to check the response for http errors here (response.status_code == 200)